### PR TITLE
Enable python3

### PR DESCRIPTION
**Pull request by plettich**
With this PR the appliance works with Python 3.
Not all functionality has been tested but the available tests run fine.
There is also a travis configuration to enable automated testing.

### CODE REVIEW

File: authappliance/menu.py [hunk 1]
```diff
@@ -498,7 +499,7 @@ def delete_tinc(self, vpn_name='privacyideaVPN'):
         self.display_messages()
 
     def display_messages(self):
-        self.d.scrollbox(self.info.decode('utf-8'), height=20, width=60)
+        self.d.scrollbox(self.info, height=20, width=60)
```
Comment (fredreichbier) [inline authappliance/menu.py]: Hm, redundancy setup fails at this line under Python 2 due to some encoding issues:
```
  File "/opt/venv/bin/pi-appliance", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/opt/appliance/authappliance/pi-appliance", line 23, in <module>
    main()
  File "/opt/appliance/authappliance/menu.py", line 2022, in main
    pS.main_menu()
  File "/opt/appliance/authappliance/menu.py", line 1789, in main_menu
    menu()
  File "/opt/appliance/authappliance/menu.py", line 727, in menu
    menu()
  File "/opt/appliance/authappliance/menu.py", line 803, in setup_redundancy
    self.peer.display_messages()
  File "/opt/appliance/authappliance/menu.py", line 502, in display_messages
    self.d.scrollbox(self.info, height=20, width=60)
  File "/opt/venv/local/lib/python2.7/site-packages/dialog.py", line 3412, in scrollbox
    f.write(text)
TypeError: write() argument 1 must be unicode, not str
```

... I guess we need something like privacyIDEA's ``to_unicode`` helper method here?

File: authappliance/menu.py [hunk 2]
```diff
@@ -541,13 +542,17 @@ def setup_redundancy(self):
 
         self.add_info("Setup my.cnf on local server...")
 
-        for key, value in shared_my_cnf_values.items() + local_my_cnf_values.items():
+        conf_values = shared_my_cnf_values.copy()
+        conf_values.update(local_my_cnf_values)
+        for key, value in conf_values:
```
Comment (fredreichbier) [inline authappliance/menu.py]: As we iterate over (key, value) pairs, I think this should be ``conf_values.items()`` :)

File: authappliance/menu.py [hunk 3]
```diff
@@ -541,13 +542,17 @@ def setup_redundancy(self):
...
             self.dbConfig.set('mysqld', key, value)
 
         self.add_info("Setup my.cnf on remote server...")
 
         remote_my_cnf = RemoteMySQLConfig(sftp)
-        for key, value in shared_my_cnf_values.items() + remote_my_cnf_values.items():
+        conf_values = shared_my_cnf_values.copy()
+        conf_values.update(remote_my_cnf_values)
+        for key, value in conf_values:
```
Comment (fredreichbier) [inline authappliance/menu.py]: same as above

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,9 @@
+netaddr==0.7.19
 argparse==1.2.1
 funcsigs==1.0.2
 paramiko==1.16.0
 pyparsing==2.0.3
-python2-pythondialog==3.0.1
+pythondialog==3.0.1
```
Comment (fredreichbier) [inline requirements.txt]: Hm, the ``pythondialog`` package breaks under Python 2.7. I see you used requirements specifiers in ``setup.py`` to install ``pythondialog`` on Python 3 and ``python2-pythondialog`` on Python 2. Can we do the same in ``requirements.txt``?

File: test/test_menu_privacyidea.py [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
+import authappliance.menu
+from authappliance.lib.appliance import PrivacyIDEAConfig
+from .mockdialog import preselected, initial
+from .testutil import MenuTestCase, ApplianceBehavior, TMP_CONFIG
+from mock import patch
 
 
 class TestMenuPrivacyIDEA(MenuTestCase):
-    def test_set_loglevel(self):
+    @patch.object(authappliance.menu.OSConfig, 'restart')
```
Comment (fredreichbier) [inline test/test_menu_privacyidea.py]: nice trick :)

Comment (plettich) [reply test/test_menu_privacyidea.py]: Thanks :-)

File: requirements.txt (hunk 4 above)
Comment (plettich) [reply requirements.txt]: I'll check that.

File: authappliance/menu.py (hunk 2 above)
Comment (plettich) [reply authappliance/menu.py]: You are right, missed that. Will fix.

File: authappliance/menu.py (hunk 1 above)
Comment (plettich) [reply authappliance/menu.py]: I guess i have to set up some kind of redundancy to test this...

File: authappliance/lib/ldap_proxy.py [hunk 6]
```diff
@@ -72,29 +72,33 @@
...
         sys.exit(1)
     return config
 
+
 class LDAPProxyConfig(object):
     def __init__(self, filename=LDAP_PROXY_CONFIG_FILE):
         self.filename = filename
         self.reset()
         self.autosave_enabled = True
+        self.config = None
```
[the remark this answers is not in the document]
Comment (plettich) [reply authappliance/lib/ldap_proxy.py]: fixed in 8f45145

File: authappliance/menu.py (hunk 1 above)
Comment (plettich) [reply authappliance/menu.py]: Unfortunately i could not reproduce this error but i fixed it anyway in 8f45145

File: authappliance/menu.py (hunk 2 above)
Comment (plettich) [reply authappliance/menu.py]: fixed in 8f45145

File: authappliance/menu.py (hunk 3 above)
Comment (plettich) [reply authappliance/menu.py]: fixed in 8f45145

File: requirements.txt (hunk 4 above)
Comment (plettich) [reply requirements.txt]: fixed in 8f45145

File: authappliance/lib/ldap_proxy.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (fredreichbier) [reply authappliance/lib/ldap_proxy.py]: Thanks!

File: authappliance/menu.py (hunk 1 above)
Comment (fredreichbier) [reply authappliance/menu.py]: Hm, I'm not sure why the error occurs for me :) But your commit fixes it, thanks!

**Pull request merged**